chandra_aca/maude_decom.py

```diff
@@ -0,0 +1,486 @@
+        img_size = data[slot_msids['sizes']]['values']
+        image = assemble_image(pixel_data, img_size)
+        # there must be an MSID to fetch this, but this works
+        subimage = np.char.replace(np.char.replace(np.char.replace(
+            img_size, '8X8', ''), '6X6', ''), '4X4', '').astype(int) - 1
+
+        result = {
+            'TIME': data[slot_msids['sizes']]['times'],
+            'IMGNUM': np.ones(len(tref), dtype='<i8') * slot,
+            'subimage': subimage,
+            'IMGSIZE': data[slot_msids['sizes']]['values'],
+            'IMGROW0': data[slot_msids['rows']]['values'],
+            'IMGCOL0': data[slot_msids['cols']]['values'],
+            'SCALE_FACTOR': data[slot_msids['scale_factor']]['values'],
+            'INTEG': 0.016 * data[msids['integration_time']]['values'],
+            'IMG': image
+        }
+
+        if adjust_time:
+            result['TIME'] -= (result['INTEG'] / 2 + 1.025)
+            # result['END_INTEG_TIME'] = result['TIME'] + result['INTEG']
+
+        if adjust_corner:
+            result['IMGROW0'][result['IMGSIZE'] == '6X61'] -= 1
+            result['IMGCOL0'][result['IMGSIZE'] == '6X61'] -= 1
+
+        if calibrate:
+            # scale specified in ACA L0 ICD, section D.2.2 (scale_factor is already divided by 32)
+            #
+            # for an incomplete images the result can look like this:
+            #     IMGSIZE SCALE_FACTOR
+            #      str4     float64
+            #     ------- ------------
+            #      8X84       nan
+            #      8X81       1.0
+            #      8X82       nan
+            #      8X83       nan
+            #      8X84       nan
+            #      8X81       1.0
+            #      8X82       nan
+            #      8X83       nan
+            #      8X84       nan
+            #      8X81       1.0
+            #
+            # so one has to set the proper scale for images 6X62, 8X82, 8X83 and 8X84.
+            # the data should cover a larger interval than requested to avoid edge effects.
+            # I also do not want to change the original array
+            # this assumes that image sizes ALWAYS alternate
+            # is there a better way to do this?
+            scale = np.array(result['SCALE_FACTOR'])
+            for name, n in [('6X61', 2), ('8X81', 4)]:
+                s1 = (result['IMGSIZE'] == name)
+                for i in range(1, n):
+                    s2 = np.roll(s1, i)  # roll and drop the ones that go over the edge
+                    s2[:i] = False
+                    scale[s2] = scale[s1][:sum(s2)]
+            result['IMG'] *= scale[:, np.newaxis, np.newaxis]
+            result['IMG'] -= 50
+
+        if full:
+            result = combine_sub_images(result)
+            del result['subimage']
+
+    return Table(result)
+
+
+def fetch(start, stop, slots=range(8), pea=1, full=False,
+          calibrate=False, adjust_time=False, adjust_corner=False):
```

Comment:
Would it make sense to accept extra kwargs to pass to maude.get_msids?  Or just CHANNEL for future ASVT work?

Reply:
I don't know if it makes sense to get the ASVT stuff in and then have it in the release to test later when we have ASVT data ... if we should add ASVT support in a future release.

Reply:
Of the available `maude.get_msids()` kwargs, I think that `channel`, `user` and `password` might need to be passed through here.  The rest are basically fixed by the requirements of how the decom is done.  So maybe we should add a `maude` kwarg that takes a dict and gets passed along as `**maude` to `maude.get_msids()`.  I don't see any reason to wait on this.

Reply:
so, you prefer one argument as opposed to two (user, password)?

Reply:
I think that was one argument instead of 3 (channel, user, password)?

Reply:
Looking at this today, probably 3 args is cleaner.